File: src/test/java/org/stt/ItemReaderTestHelper.java
package org.stt;

import org.mockito.BDDMockito.BDDMyOngoingStubbing;
import org.stt.model.TimeTrackingItem;
import org.stt.persistence.ItemReader;

import java.util.Optional;

import static org.mockito.BDDMockito.given;

public class ItemReaderTestHelper {

	public static void givenReaderReturns(ItemReader itemReader,
			TimeTrackingItem... items) {

		BDDMyOngoingStubbing<Optional<TimeTrackingItem>> stubbing = given(itemReader
				.read());
		for (TimeTrackingItem item : items) {
			stubbing = stubbing.willReturn(Optional.of(item));
		}
        stubbing.willReturn(Optional.empty());
    }
}

File: src/test/kotlin/org/stt/ItemReaderTestHelper.kt
package org.stt

import org.mockito.BDDMockito.BDDMyOngoingStubbing
import org.mockito.BDDMockito.given
import org.stt.model.TimeTrackingItem
import org.stt.persistence.ItemReader

object ItemReaderTestHelper {

    fun givenReaderReturns(itemReader: ItemReader,
                           vararg items: TimeTrackingItem) {

        var stubbing: BDDMyOngoingStubbing<TimeTrackingItem> = given(itemReader
                .read())
        for (item in items) {
            stubbing = stubbing.willReturn(item)
        }
        stubbing.willReturn(null)
    }
}
